File: 1710 - Maximum Units on a Truck/solution.py
from operator import itemgetter
class Solution:
    def maximumUnits(self, A: List[List[int]], sz: int) -> int:
        A.sort(key=itemgetter(1), reverse=True)
        res = 0
        for i, j in A:
            if sz == 0:
                break
            res += j*min(sz, i)
            sz -= min(sz, i)
        return res

File: 1710 - Maximum Units on a Truck/solution.c
#define min(a,b) (((a)<(b))?(a):(b))
int maximumUnits(int** A, int n, int* colSz, int sz){
    int bucket[1001] = {}, res = 0;
    for(int i = 0; i < n; i++) bucket[A[i][1]] += A[i][0];
    for(int i = 1000; i > 0; i--){
        if(sz == 0) break;
        res += i*min(sz, bucket[i]);
        sz -= min(sz, bucket[i]);
    }
    return res;
}